test/e2e/kubectl.go

```diff
@@ -56,49 +48,131 @@ var _ = Describe("kubectl", func() {
+			defer cleanup(nautilusPath, updateDemoSelector)
+
+			By("creating a replication controller")
+			runKubectl("create", "-f", nautilusPath)
+			validateController(c, nautilusImage, 2)
+			By("scaling down the replication controller")
+			runKubectl("resize", "rc", "update-demo-nautilus", "--replicas=1")
+			validateController(c, nautilusImage, 1)
+			By("scaling up the replication controller")
+			runKubectl("resize", "rc", "update-demo-nautilus", "--replicas=2")
+			validateController(c, nautilusImage, 2)
+		})
+
+		It("should do a rolling update of a replication controller", func() {
+			// Cleanup all resources in case we fail somewhere in the middle
+			defer cleanup(updateDemoRoot, updateDemoSelector)
+
+			By("creating the initial replication controller")
+			runKubectl("create", "-f", nautilusPath)
+			validateController(c, nautilusImage, 2)
+			By("rollingupdate to new replication controller")
+			runKubectl("rollingupdate", "update-demo-nautilus", "--update-period=1s", "-f", kittenPath)
+			validateController(c, kittenImage, 2)
+		})
 	})
 
-	It("should scale a replication controller", func() {
-		defer cleanup(nautilusPath)
-
-		By("creating a replication controller")
-		runKubectl("create", "-f", nautilusPath)
-		validateController(c, nautilusImage, 2)
-		By("scaling down the replication controller")
-		runKubectl("resize", "rc", "update-demo-nautilus", "--replicas=1")
-		validateController(c, nautilusImage, 1)
-		By("scaling up the replication controller")
-		runKubectl("resize", "rc", "update-demo-nautilus", "--replicas=2")
-		validateController(c, nautilusImage, 2)
-	})
+	Describe("guestbook", func() {
+		var (
+			guestbookPath       = filepath.Join(testContext.repoRoot, "examples/guestbook")
+			frontendSelector    = "name=frontend"
+			redisMasterSelector = "name=redis-master"
+			redisSlaveSelector  = "name=redis-slave"
+		)
+
+		It("should create and stop a working application", func() {
+			defer cleanup(guestbookPath, frontendSelector, redisMasterSelector, redisSlaveSelector)
 
-	It("should do a rolling update of a replication controller", func() {
-		// Cleanup all resources in case we fail somewhere in the middle
-		defer cleanup(updateDemoRoot)
+			By("creating all guestbook components")
+			runKubectl("create", "-f", guestbookPath)
 
-		By("creating the initial replication controller")
-		runKubectl("create", "-f", nautilusPath)
-		validateController(c, nautilusImage, 2)
-		By("rollingupdate to new replication controller")
-		runKubectl("rollingupdate", "update-demo-nautilus", "--update-period=1s", "-f", kittenPath)
-		validateController(c, kittenImage, 2)
+			By("validating guestbook app")
+			validateGuestbookApp(c)
+		})
 	})
 
 })
 
-func cleanup(filePath string) {
+func validateGuestbookApp(c *client.Client) {
+	// Wait 2 minutes for frontend to serve content.
+	serving := false
+	for i := 1; i < 24; i++ {
+		entry, err := makeRequestToGuestbook(c, "get", "")
+		if err == nil && entry == "{\"data\": \"\"}" {
```

Comment:
Ugh, github ate my formatting. Use ` instead of " for the outer quotes and you won't need to escape the quotes in the json literal. See http://play.golang.org/p/KRTIqzHJLl if that didn't come across well enough in text.